src/main/scala/is/hail/expr/AST.scala

```diff
@@ -307,6 +310,17 @@ case class Const(posn: Position, value: Any, t: Type) extends AST(posn) {
   })
 
   override def typecheckThis(): Type = t
+
+  def toIR: Option[IR] = (t, value) match {
```

Comment:
I'm not going to say change this, but do we think this is clearer than:

```
  def toIROpt: Option[IR] = {
    try {
      Some(toIR)
    } except {
      case ASTToIRConversionException => None
    }
  }

  def toIR: IR // might fail with ASTToIRConversionException
```

In particular, the parital-ness of this method is temporary (the entire AST needs to be convertable to the IR), although the AST itself is temporary (albeit on a longer timescale).

Reply:
Either one seems fine to me. Exceptions are certainly _The Java Way_. Probably compile better too.

Reply:
`PartialFunction` would need to be injected into that exception some how, or we'd have to use whatever exception it throws (and make sure we don't accidentally throw that exception for an unrelated reason).

Reply:
Yeah, I was thinking there'd be an exception for just for the partialness of this function.